security-policies/bundle/compliance/cis_azure/rules/cis_3_11/rule.rego

```diff
@@ -0,0 +1,31 @@
+package compliance.cis_azure.rules.cis_3_11
+
+import data.compliance.lib.common
+import data.compliance.policy.azure.data_adapter
+import future.keywords.if
+
+finding = result if {
+	# filter
+	data_adapter.is_storage_account
+	_ = data_adapter.resource.extension.blobService
+
+	# set result
+	result := common.generate_result_without_expected(
+		common.calculate_result(ensure_soft_delete_is_enabled),
+		{"Resource": data_adapter.resource},
+	)
+}
+
+default ensure_soft_delete_is_enabled = false
+
+ensure_soft_delete_is_enabled if {
+	is_policy_valid(data_adapter.resource.extension.blobService.deleteRetentionPolicy)
+	data_adapter.resource.extension.blobService.deleteRetentionPolicy.allowPermanentDelete == false
+	is_policy_valid(data_adapter.resource.extension.blobService.containerDeleteRetentionPolicy)
+}
+
+is_policy_valid(policy) if {
+	policy.enabled == true
+	policy.days >= 7
```

Comment:
When reading the policy it seems to me the only check needed is if days isn't empty or null. It's mentioned the 7 to 365 rule, but in the checks just say:
<img width="1296" alt="image" src="https://github.com/elastic/cloudbeat/assets/5350001/1d4b15af-ccb8-46fb-af0f-7e06ef35a81f">

Reply:
Indeed, I followed the text on the Rationale part. I am not sure what would be the correct thing to do here.

Reply:
Maybe quick discussion with the team? Not sure either. I would implement the steps in the Audit part, to avoid false positives. But also, I have been doing this for 1 month. Better to get a second opinion

Reply:
Especially the less than 365 days part seems fishy

Reply:
So, you also suggest following the steps as described in the audit. Ok, I will update the rule check then.

Reply:
Replaced it with `days > 0`